### PR TITLE
CVL2 + more examples

### PR DESCRIPTION
**Pull request by ghilac**
Adding CVL2 updates examples and adding another example

### CODE REVIEW

File: certora/conf/ERC20.conf [hunk 1]
```diff
@@ -1,15 +1,15 @@
...
   "optimistic_loop": false,
   "multi_assert_check": false,
   "send_only": true,
-  "disableLocalTypeChecking": true,
-  "process": "emv"
-}
\ No newline at end of file
+  "process": "emv",
+  "server": "production",
+  "prover_version": "master"
```
Comment (shellygr) [inline certora/conf/ERC20.conf]: can we omit this?

File: certora/conf/ERC20.conf [hunk 2]
```diff
@@ -1,15 +1,15 @@
...
   "msg": "ERC20",
   "optimistic_loop": false,
   "multi_assert_check": false,
   "send_only": true,
-  "disableLocalTypeChecking": true,
-  "process": "emv"
-}
\ No newline at end of file
+  "process": "emv",
+  "server": "production",
```
Comment (shellygr) [inline certora/conf/ERC20.conf]: I think this can be omitted too

File: certora/spec/ERC20.spec [hunk 3]
```diff
@@ -72,6 +72,19 @@ rule transferDoesntRevert {
...
+   uint256 balanceRecipient = balanceOf(e, recipient);
+   transfer(e, recipient, amount);
+
+	assert balanceRecipient+ balanceSender == 
+        balanceOf(e, e.msg.sender) + balanceOf(e, recipient),
+        "Transfer of money should not affect the total sum of balances";
+}
+
 rule transferFromSpec {
     // TODO!
```
Comment (shellygr) [inline certora/spec/ERC20.spec]: not really on you Ghila, but is this todo on purpose?

File: certora/conf/ERC20.conf [hunk 4]
```diff
@@ -1,15 +1,15 @@
 {
   "files": [
-    "contracts/ERC20.sol:ERC20"
-  ],
-  "verify": [
-    "ERC20:certora/spec/ERC20.spec"
+    "contracts/ERC20TransferBug.sol:ERC20"
```
Comment (shellygr) [inline certora/conf/ERC20.conf]: oh I think you might want a separate conf?

**Review by shellygr — COMMENTED** [review]
see my comments. I may lack some context on where this example ended up after I touched it...

File: certora/spec/ERC20.spec (hunk 3 above)
Comment (ghilac) [reply certora/spec/ERC20.spec]: I removed all the unhandled ones for now

File: certora/conf/ERC20.conf (hunk 4 above)
Comment (ghilac) [reply certora/conf/ERC20.conf]: yes sorry my bad, I forgot I modified the original one

**Review by shellygr — APPROVED** [review]
thanks for taking care of this!

File: certora/conf/ERC20.conf (hunk 2 above)
Comment (ghilac) [reply certora/conf/ERC20.conf]: right

**Pull request merged**